pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -318,6 +331,10 @@ func checkCertValidity(certBytes, keyBytes []byte, ipAddresses []string, nodeUID
 	// Check that the cert is valid for the provided ip addresses
 	errs := []error{}
 	for _, ipAddress := range ipAddresses {
+		// This will skip checking ipv6 addresses as in dual-stack mode, this will cause etcd operator go degraded.
+		if isDualStackMode(ipAddresses) && net.IsIPv6String(ipAddress) {
```

Comment:
Can we cache the result of isDualStackModel() so that we're not re-invoking it for each element of the ipAddresses list please?

Reply:
Also, I just found that k8s.io/utils/net has a function [IsDualStackIPStrings] that we could use here instead of the custom function.

[IsDualStackIPStrings]: https://pkg.go.dev/k8s.io/utils/net#IsDualStackIPStrings

Reply:
@dusk125 the list is just two items :-)

good catch @ the isDualStackIPStrings, I haven't noticed it there.